source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift

```diff
@@ -84,44 +84,85 @@ class TextInputRenderer: NSObject, BaseCardElementRendererProtocol {
+            acrView.addInputHandler(textView)
         }
         if renderButton {
             stackview.addArrangedSubview(textView)
-            addInlineButton(parentview: stackview, view: textView, element: inputBlock, style: style, with: hostConfig)
+            addInlineButton(parentview: stackview, view: textView, element: inputBlock, style: style, with: hostConfig, rootview: rootView)
             return stackview
         }
         return textView
     }
-    private func addInlineButton(parentview: NSStackView, view: NSView, element: ACSTextInput, style: ACSContainerStyle, with hostConfig: ACSHostConfig) {
+    private func addInlineButton(parentview: NSStackView, view: NSView, element: ACSTextInput, style: ACSContainerStyle, with hostConfig: ACSHostConfig, rootview: NSView) {
         let action = element.getInlineAction()
-        let button = NSButton(title: action?.getTitle() ?? "", target: self, action: Selector(("test")))
+        let button = ACRButton(style: .inline)
+        button.title = action?.getTitle() ?? ""
+        button.cornerRadius = 0
+        button.isBordered = false
+        
         let attributedString: NSMutableAttributedString
         attributedString = NSMutableAttributedString(string: button.title)
         if let colorHex = hostConfig.getForegroundColor(style, color: .default, isSubtle: true), let textColor = ColorUtils.color(from: colorHex) {
             attributedString.addAttributes([.foregroundColor: textColor], range: NSRange(location: 0, length: attributedString.length))
         }
         parentview.addArrangedSubview(button)
         button.translatesAutoresizingMaskIntoConstraints = false
-        button.isBordered = false
+        
         if view is ACRMultilineInputTextView {
             button.setContentHuggingPriority(.required, for: .vertical)
             button.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
         }
         button.widthAnchor.constraint(lessThanOrEqualTo: view.widthAnchor, multiplier: 0.5).isActive = true
         button.attributedTitle = attributedString
+        
         // image icon
         if let imageIcon = action?.getIconUrl(), !imageIcon.isEmpty {
             guard let url = URL(string: imageIcon) else { return }
             DispatchQueue.global().async {
                 guard let data = try? Data(contentsOf: url) else { return }
                 DispatchQueue.main.async {
-                    let image = NSImage(data: data)
-                    image?.size = .init(width: button.bounds.width, height: button.bounds.height)
+                    guard let image = NSImage(data: data) else {
+                        return
+                    }
+                    image.size = .init(width: button.bounds.width, height: button.bounds.height)
+                    button.showsIcon = true
+                    button.title = ""
+                    button.iconImageSize = NSSize(width: image.size.width * 0.5, height: image.size.height)
```

Comment:
Have this as constant. Size shouldn't depend on image size as it maybe anything like 1080x720
```suggestion
                    button.iconImageSize = NSSize(width: 20, height: 20)
```